soh/include/z64save.h

```diff
@@ -83,127 +128,145 @@ typedef struct {
+    /* 0x003D */ u8 isDoubleDefenseAcquired;
     /* 0x003E */ u8 bgsFlag;
     /* 0x003F */ u8 ocarinaGameRoundNum;
     /* 0x0040 */ ItemEquips childEquips;
     /* 0x004A */ ItemEquips adultEquips;
-    /* 0x0054 */ u32 unk_54; // this may be incorrect, currently used for alignement
-    /* 0x0066 */ s16 savedSceneNum;
+    /* 0x0054 */ u32 unk_54; // this may be incorrect, currently used for alignment
+    /* 0x0058 */ char unk_58[0x0E];
+    /* 0x0066 */ s16 savedSceneNum; // Upstream TODO: sceneId
     /* 0x0068 */ ItemEquips equips;
     /* 0x0074 */ Inventory inventory;
     /* 0x00D4 */ SavedSceneFlags sceneFlags[124];
     /* 0x0E64 */ FaroresWindData fw;
+    /* 0x0E8C */ char unk_E8C[0x10];
     /* 0x0E9C */ s32 gsFlags[6];
+    /* 0x0EB4 */ char unk_EB4[0x4];
     /* 0x0EB8 */ s32 highScores[7];
     /* 0x0ED4 */ u16 eventChkInf[14]; // "event_chk_inf"
     /* 0x0EF0 */ u16 itemGetInf[4]; // "item_get_inf"
     /* 0x0EF8 */ u16 infTable[30]; // "inf_table"
+    /* 0x0F34 */ char unk_F34[0x04];
     /* 0x0F38 */ u32 worldMapAreaData; // "area_arrival"
-    /* 0x0F40 */ u8 scarecrowCustomSongSet;
-    /* 0x0F41 */ OcarinaNote scarecrowCustomSong[108];
+    /* 0x0F3C */ char unk_F3C[0x4];
+    /* 0x0F40 */ u8 scarecrowLongSongSet;
+    /* 0x0F41 */ OcarinaNote scarecrowLongSong[108]; // Upstream TODO: Audio
+    /* 0x12A1 */ char unk_12A1[0x24];
     /* 0x12C5 */ u8 scarecrowSpawnSongSet;
-    /* 0x12C6 */ OcarinaNote scarecrowSpawnSong[16];
+    /* 0x12C6 */ OcarinaNote scarecrowSpawnSong[16]; // Upstream TODO: Audio
     /* 0x1346 */ char unk_1346[0x02];
     /* 0x1348 */ HorseData horseData;
+    /* 0x1352 */ u16 checksum; // "check_sum"
     /* 0x1354 */ s32 fileNum; // "file_no"
     /* 0x1358 */ char unk_1358[0x0004];
     /* 0x135C */ s32 gameMode;
-    /* 0x1360 */ s32 sceneSetupIndex;
+    /* 0x1360 */ s32 sceneSetupIndex; // "counter" // Upstream TODO: sceneLayer
     /* 0x1364 */ s32 respawnFlag; // "restart_flag"
-    /* 0x1368 */ RespawnData respawn[3]; // "restart_data"
+    /* 0x1368 */ RespawnData respawn[RESPAWN_MODE_MAX]; // "restart_data"
     /* 0x13BC */ f32 entranceSpeed;
     /* 0x13C0 */ u16 entranceSound;
-    /* 0x13C3 */ u8 unk_13C3;
+    /* 0x13C2 */ char unk_13C2[0x0001];
+    /* 0x13C3 */ u8 retainWeatherMode;
     /* 0x13C4 */ s16 dogParams;
     /* 0x13C6 */ u8 textTriggerFlags;
     /* 0x13C7 */ u8 showTitleCard;
     /* 0x13C8 */ s16 nayrusLoveTimer;
+    /* 0x13CA */ char unk_13CA[0x0002];
     /* 0x13CC */ s16 rupeeAccumulator;
     /* 0x13CE */ s16 timer1State;
     /* 0x13D0 */ s16 timer1Value;
     /* 0x13D2 */ s16 timer2State;
     /* 0x13D4 */ s16 timer2Value;
     /* 0x13D6 */ s16 timerX[2];
     /* 0x13DA */ s16 timerY[2];
+    /* 0x13DE */ char unk_13DE[0x0002];
     /* 0x13E0 */ u8 seqId;
     /* 0x13E1 */ u8 natureAmbienceId;
-    /* 0x13E2 */ u8 buttonStatus[9];
+    // #region SOH [Enhancements]
+    // Support Dpad equips
+    /* 0x13E2 */ u8 buttonStatus[5];
```

Comment:
The `5` here should not have been changed
```suggestion
    /* 0x13E2 */ u8 buttonStatus[9];
```